docs/proposals/20250325-distributed-inference.md

```diff
@@ -133,35 +132,135 @@ env:
   - name: POD_INDEX
     valueFrom:
       fieldRef:
-        fieldPath: metadata.annotations['apps.kubernetes.io/pod-index']
+        fieldPath: metadata.labels['apps.kubernetes.io/pod-index']
 ...
 ```
 
 ### Liveness and Readiness Probes
 
-Standard HTTP probes are insufficient for multi-node vLLM, as only the leader pod serves the `/health` endpoint at port 5000. Probes must be adapted to verify both Ray cluster integrity (all nodes) and the leader's vLLM server readiness.
+Standard HTTP probes are insufficient for multi-node vLLM, as only the leader pod serves the /health endpoint at port 5000, and Ray cluster dynamics in Kubernetes introduce significant complexities. Key challenges include:
+
+- **Leader Dependency and Initialization**: The leader waits for all n-1 worker pods to join the Ray cluster (where n is the StatefulSet replica count) before starting tasks like model downloading and loading it to GPU memory. If a worker pod fails permanently during initialization or restarts later, the leader does not reinitialize the new worker, rendering the vLLM service unusable until the leader is restarted to recreate the Ray cluster from a clean state.
+- **Worker Rejoining Delay:** In the case of a leader restart, existing worker pods may take 1–2 minutes to recognize a new leader, as they may continue sending heartbeats to the old leader.
```

Comment:
From my test, it cost only 10seconds to let worker rejoin to the ray cluster after leader restarting. 
Let's make thing clearer. There are two phases of initialization. 1. ray cluster initialization. It cost about 10seconds 2. vllm application initialization. It cost couple minutes.

worker rejoin in the phase1 works, it cost couple of seconds.
worker rejoin in the phase2 don't work.